source/utils/adios_remote_read/adios_remote_read.cpp

```diff
@@ -0,0 +1,721 @@
+                        startX = countX * rank;
+
+                        if (rank == nproc - 1)
+                        {
+                            // last process need to read all the rest of slices
+                            countX = globalSizeX - countX * (nproc - 1);
+                        }
+                        break;
+                    case DIM3Y:
+                        countY = globalSizeY / nproc;
+                        startY = countY * rank;
+
+                        if (rank == nproc - 1)
+                        {
+                            // last process need to read all the rest of slices
+                            countY = globalSizeY - countY * (nproc - 1);
+                        }
+                        break;
+                    case DIM3Z:
+                        countZ = globalSizeZ / nproc;
+                        startZ = countZ * rank;
+
+                        if (rank == nproc - 1)
+                        {
+                            // last process need to read all the rest of slices
+                            countZ = globalSizeZ - countZ * (nproc - 1);
+                        }
+                        break;
+                    default:
+                        break;
+                    }
+                    var.SetSelection(adios2::Box<adios2::Dims>({startX, startY, startZ},
+                                                               {countX, countY, countZ}));
+                    size_t elementsSize = var.SelectionSize();
+                    std::vector<double> data3D(elementsSize);
+                    reader.Get<double>(var, data3D.data(), adios2::Mode::Sync);
+                    if (DEBUG)
+                        out = getOutputString(rank, output_line_length, out, data3D);
+                }
+            }
+
+            reader.EndStep();
+        }
+        reader.Close();
+        end_time = std::chrono::system_clock::now();
+        std::chrono::duration<double> elapsed_seconds = end_time - start_time;
+        out += "Duration: " + std::to_string(elapsed_seconds.count()) + "\n";
+        std::ofstream out_file;
+        out_file.open("output3D-" + std::to_string(rank) + ".log");
+        out_file << out;
+        out_file.close();
+    }
+    catch (std::invalid_argument &e)
+    {
+        if (rank == 0)
+        {
+            std::cout << "Invalid argument exception, STOPPING PROGRAM\n";
+            std::cout << e.what() << "\n";
+        }
+    }
+    catch (std::ios_base::failure &e)
+    {
+        if (rank == 0)
+        {
+            std::cout << "System exception, STOPPING PROGRAM\n";
+            std::cout << e.what() << "\n";
+        }
+    }
+    catch (std::exception &e)
+    {
+        if (rank == 0)
+        {
+            std::cout << "Exception, STOPPING PROGRAM\n";
+            std::cout << e.what() << "\n";
+        }
+    }
+}
+
+void read3DPlane(int nproc, int rank, const std::string &filename, const int nsteps, adios2::IO &io,
```

Comment:
You might want to add a comment for each of these functions so that it's clear what you are testing. In this case it seems you are bringing the XZ plane (with X data divided across ranks) and only a fixed ration of the Z data (using the weird start equation as starting point)